test/fake_cloud_firestore_test.dart

```diff
@@ -683,6 +684,44 @@ void main() {
       expect(timeDiff, lessThan(1000));
     });
 
+    test('FieldValue.serverTimestamp() sets the time with fixed time',
+        () async {
+      final fixedTimestamp = Timestamp.fromMicrosecondsSinceEpoch(100);
+      final timeProvider = FixedServerTimeProvider(fixedTimestamp);
+      final firestore = FakeFirebaseFirestore(
+        fakeServerTimeProvider: timeProvider,
+      );
+      await firestore.collection('users').doc(uid).set({
+        'created': FieldValue.serverTimestamp(),
+      });
+      final users = await firestore.collection('users').get();
+      final bob = users.docs.first;
+      expect(bob.get('created'), isNotNull);
+      final bobCreated = bob.get('created') as Timestamp;
+      expect(bobCreated, fixedTimestamp);
+    });
+
+    test('FieldValue.serverTimestamp() sets the time with relative time',
```

Comment:
If we use Clock, the first test is probably sufficient and you can remove the second test. If you want to implement a second test, perhaps you could use [fake_async](https://pub.dartlang.org/packages/fake_async) and demonstrate the passing of time like they do in https://pub.dev/packages/clock.

Reply:
I think that will be like:
```dart
final fakeClock = Clock(() => DateTime.now().subtract(Duration(hours: 3)));
```

So, maybe I don't prefer to have this test just for demonstration and probably mess the tests up.
What do you think?🤔